client/session.go

```diff
@@ -170,8 +192,38 @@ func (s *session) setTopologyMap(topologyMap m3db.TopologyMap) error {
 	s.Lock()
 	prev := s.queues
 	s.queues = queues
-	s.setWriteOpPoolsWithLock(topologyMap)
 	s.topoMap = topologyMap
+	prevReplicas := s.replicas
+	s.replicas = topologyMap.Replicas()
+	s.quorum = topologyMap.QuorumReplicas()
+	if s.fetchBatchOpArrayArrayPool == nil ||
+		s.fetchBatchOpArrayArrayPool.Entries() != len(queues) {
+		s.fetchBatchOpArrayArrayPool = newFetchBatchOpArrayArrayPool(
+			s.opts.GetFetchBatchOpPoolSize(),
+			len(queues),
+			s.opts.GetFetchBatchOpPoolSize()/s.opts.GetFetchBatchSize())
```

Comment:
Signature is: 

```go
newFetchBatchOpArrayArrayPool(size int, entries int, capacity int)
```

`size` is the count of entries in the pool, i.e. how many batches you want to concurrently execute, this is directly what you set as the size of the pool as.
`entries` is the size that each top level array is returned every time we get an element from the pool, we need an entry in the array for each host queue so this is always sized to `len(queues)`.
`capacity` is the amount of of batch ops that can be put into each host queue, considering that all IDs could randomly shard to the same host this should be `maxAmountOfFetchesIntendedToAskForAtOnce / fetchBatchSize`.  

I think for lack of actually defining "maxAmountOfFetchesIntendedToAskForAtOnce" in the options I chose "GetFetchBatchOpPoolSize()", not sure why though.  Should either add this as a param or just choose a fixed number I suppose.  For the sake of discussion, say one chooses to fetch at most 10k at once and batch size is 128, then it should be 78.125 so that's about roughly the size we should look for.